### PR TITLE
Feature/parallel batch (depends on feature/gpus)

### PR DESCRIPTION
**Pull request by bmcdanie**
parallel batch building with optional openmp. time advance 2x+ speedup for large problems.

### CODE REVIEW

**Comment by grahamlopez** [comment]
okay to test

File: src/batch.cpp [hunk 1]
```diff
@@ -1,4 +1,9 @@
 #include "batch.hpp"
+
+#ifdef ASGARD_USE_OPENMP
```
Comment (grahamlopez) [inline src/batch.cpp]: follow the pattern for `ASGARD_IO_HIGHFIVE` here to get the conditional build info

File: src/batch.cpp [hunk 2]
```diff
@@ -577,10 +582,28 @@ build_batches(PDE<P> const &pde, element_table const &elem_table,
   std::vector<batch_operands_set<P>> batches =
       allocate_batches<P>(pde, elements_in_chunk);
 
-  // loop over elements
-  // FIXME eventually want to do this in parallel
-  for (const auto &[i, connected] : chunk)
+  std::vector<int> const index_to_key = [&chunk]() {
```
Comment (grahamlopez) [inline src/batch.cpp]: the need for this indirection is not yet obvious to me.

Comment (bmcdanie) [reply src/batch.cpp]: I really want to loop over `[i, connected]` as I did before, but I could not find a way to make the structured binding work with openmp. `i` may not start at zero, and although our current chunking implementation assures that the range of `i` is contiguous, I didn't want to rely on that.

so, instead, I am forming a range from 0 to `chunk.size()` to loop over, and then retrieving the associated `[i, connected]` values inside the loop.

File: src/batch.cpp (hunk 1 above)
Comment (bmcdanie) [reply src/batch.cpp]: I patterned this on how `ASGARD_IO_HIGHFIVE` is set - or intended to. what changes would you like here?

Comment (bmcdanie) [reply src/batch.cpp]: changed to autogenerated header definition per Slack convo

**Pull request merged**